### PR TITLE
Fix `float` marshal, unmarshall functions

### PR DESCRIPTION
**Pull request by illia-li**
Changes:
1. On unmarshall data which length different from 0 or 4 was `no error` before, now return an error.
2. Optimized marshal, unmarshall functions.

### CODE REVIEW

File: marshal.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"encoding/binary"
 	"errors"
 	"fmt"
+	"github.com/gocql/gocql/serialization/float"
```
Comment (sylwiaszunejko) [inline marshal.go:12]: it should be below with other gocql/serialization imports

File: serialization/float/marshal.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"reflect"
+)
+
+func Marshal(value interface{}) ([]byte, error) {
+	switch v := value.(type) {
+	case nil:
+		return nil, nil
+
```
Comment (sylwiaszunejko) [inline serialization/float/marshal.go:11]: is this new line needed?

File: serialization/float/unmarshal.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"fmt"
+	"reflect"
+)
+
+func Unmarshal(data []byte, value interface{}) error {
+	switch v := value.(type) {
+	case nil:
+		return nil
+
```
Comment (sylwiaszunejko) [inline serialization/float/unmarshal.go:12]: unnecessary new line

**Review by sylwiaszunejko — CHANGES_REQUESTED** [review]

**Review by dkropachev — CHANGES_REQUESTED** [review]
@illia-li , all good, please get rid of empty lines

File: marshal.go (hunk 1 above)
Comment (illia-li) [reply marshal.go]: Done.

File: serialization/float/marshal.go (hunk 2 above)
Comment (illia-li) [reply serialization/float/marshal.go]: Deleted

File: serialization/float/unmarshal.go (hunk 3 above)
Comment (illia-li) [reply serialization/float/unmarshal.go]: Deleted

**Review by sylwiaszunejko — APPROVED** [review]

**Review by dkropachev — APPROVED** [review]

**Pull request merged**